### PR TITLE
C1

### PR DESCRIPTION
**Pull request by Bravo-inc**

### CODE REVIEW

File: src/main/java/mk/p/singleton/v1/MySingleton.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package mk.p.singleton.v1;
+
+public class MySingleton {
+    private MySingleton() {
+        System.out.println("Singleton created");
+    }
+
+    public static final MySingleton INSTANCE = new MySingleton();
```
Comment (olezhik) [inline src/main/java/mk/p/singleton/v1/MySingleton.java]: Why this field is defined between methods?

File: src/main/java/mk/p/singleton/v1/Test.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package mk.p.singleton.v1;
+
+/**
+ * Simply
```
Comment (olezhik) [inline src/main/java/mk/p/singleton/v1/Test.java]: What's the purpose of this comment?

File: src/main/java/mk/p/singleton/v4/MySingleton.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package mk.p.singleton.v4;
+
+public class MySingleton {
+    private MySingleton() {
+        System.out.println("Singleton created");
+    }
+
+    private static class SingletonHolder {
+        public static final MySingleton HOLDER_INSTANCE = new MySingleton();
+    }
```
Comment (olezhik) [inline src/main/java/mk/p/singleton/v4/MySingleton.java]: Why the inner class definition is between methods of the parent class?

**Review by olezhik — CHANGES_REQUESTED** [review]

File: src/main/java/mk/p/singleton/v1/Test.java (hunk 2 above)
Comment (Bravo-inc) [reply src/main/java/mk/p/singleton/v1/Test.java]: This comment is for me.

File: src/main/java/mk/p/singleton/v4/MySingleton.java (hunk 3 above)
Comment (Bravo-inc) [reply src/main/java/mk/p/singleton/v4/MySingleton.java]: Why not?

File: src/main/java/mk/p/singleton/v1/MySingleton.java (hunk 1 above)
Comment (Bravo-inc) [reply src/main/java/mk/p/singleton/v1/MySingleton.java]: Coz it is a new style.

File: src/main/java/mk/p/singleton/v4/MySingleton.java (hunk 3 above)
Comment (olezhik) [reply src/main/java/mk/p/singleton/v4/MySingleton.java]: Review java code convention doc

File: src/main/java/mk/p/singleton/v1/MySingleton.java (hunk 1 above)
Comment (olezhik) [reply src/main/java/mk/p/singleton/v1/MySingleton.java]: Review java code convention doc

**Review by olezhik — APPROVED** [review]

**Pull request merged**